### ISSUE TITLE
Wrong formatting rule for pattern matching function

### ISSUE THREAD

**Issue by Volcomix** [fsprojects/FSharpLint#298]
### Description

Wrong formatting rule for pattern matching function.

### Repro steps

```fsharp
1 |> function
    | 1 -> true // 4 spaces indentation
    | _ -> false
```

### Expected behavior

Should not generate a warning as it seems to respect [that style-guide](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#pattern-matching-constructs) (see the 3rd and 4th examples).

### Actual behavior

Generate a warning: `Lint: Match clauses should be indented one level from 'function' keyword.`

### Known workarounds

We can get rid of the warning by aligning the `1` and the `_` with the start of `function`:

```fsharp
1 |> function
   | 1 -> true // 3 spaces indentation
   | _ -> false
```
or
```fsharp
1 |> (function
    | 1 -> true // 4 spaces indentation
    | _ -> false)
```
The last one is close to the [unit test](https://github.com/fsprojects/FSharpLint/blob/a187372a5c4ab0d6481031ae6ba53e712018c056/tests/FSharpLint.Core.Tests/Rules/TestFormattingRules.fs#L667) of the invoked rule.
